### PR TITLE
Break minichef and migrations out to their own fns

### PR DESCRIPTION
**Pull request by hammeiam**
The first of several steps to remove logic from usePoolData and separate concerns into their own files. Notice I'm also moving from poolname to address

### CODE REVIEW

**Review by nkrulikovsky — APPROVED** [review]
Nice using of Tuple. 
LGTM!

File: src/utils/minichef.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    ) as MulticallContract<MiniChef>
+
+    const [saddlePerSecond, totalAllocPoint] = await ethCallProvider.tryEach(
+      [minichefContract.saddlePerSecond(), minichefContract.totalAllocPoint()],
+      [false, false],
+    )
+
+    const poolInfos = await ethCallProvider.tryEach(
+      addressesPidTuples.map(([, pid]) => minichefContract.poolInfo(pid)),
+      Array(addressesPidTuples.length).fill(true),
```
Comment (ug02fast) [inline src/utils/minichef.ts:44]: I was actually trying to remember this style of filling arrays when I was refactoring multicalls to `tryEach`

**Review by ug02fast — APPROVED** [review]
tuple199fan

File: src/utils/minichef.ts (hunk 1 above)
Comment (hammeiam) [reply src/utils/minichef.ts:44]: I feel like `calls.map(() => true)` is maybe shorter, but it all does the same thing

File: src/utils/migrations.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      const result = results[i]?.newPoolAddress
+      if (result && result !== AddressZero) {
+        return { [address]: result, ...acc }
+      }
+      return acc
+    }, {} as MigrationData)
+    return parsedData
+  } catch (e) {
+    const error = new Error(`Failed to get migration data`)
+    error.stack = (e as Error).stack
```
[the remark this answers is not in the document]
Comment (hammeiam) [reply src/utils/migrations.ts]: good call out, I changed it

**Pull request merged**